### ISSUE TITLE
PHP 8.0 support

### ISSUE THREAD

**Issue by Ghostscypher** [beyondcode/laravel-websockets#639]
Please add support for ^PHP8.0, issues encountered while upgrading to PHP8.0

**Comment by myckhel** [comment]
Which lar-websockets version did you use?
V1.9 or v2-beta.

V2 has php8 support

**Comment by Ghostscypher** [comment]
I tried to use V2 beta but it breaks saying undefined constant SIGTERM, running v1.9 with php 8 seems to work fine so far

**Comment by myckhel** [comment]
You could have pasted your errors here so contributors could help out.

**Comment by jeovajr** [comment]
I'm getting this error on PHP 8.0 and Laravel 8. Using the V2.

```
> php artisan websockets:serve

   Error

  Undefined constant "BeyondCode\LaravelWebSockets\Console\Commands\SIGTERM"

  at ...\vendor\beyondcode\laravel-websockets\src\Console\Commands\StartServer.php:175
    171▕         // signal, it should mark the server as unavailable
    172▕         // to receive new connections, close the current connections,
    173▕         // then stopping the loop.
    174▕
  ➜ 175▕         $this->loop->addSignal(SIGTERM, function () {
    176▕             $this->line('Closing existing connections...');
    177▕
    178▕             $this->triggerSoftShutdown();
    179▕         });

  1   ...\vendor\beyondcode\laravel-websockets\src\Console\Commands\StartServer.php:82
      BeyondCode\LaravelWebSockets\Console\Commands\StartServer::configurePcntlSignal()

  2   ...\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:36
      BeyondCode\LaravelWebSockets\Console\Commands\StartServer::handle()
```

**Comment by Ghostscypher** [comment]
@jeovajr yea exactly the error I had. @myckhel please check it out. Thanks in advance.

**Comment by jeovajr** [comment]
I'm having this issue only on Windows.

It seems to be related with the extension pcntl.

Since it's just a test environment, I added a check for the extension on the **handle** method of the `...\vendor\beyondcode\laravel-websockets\src\Console\Commands\StartServer.php` class.

Basically:

```
        if (extension_loaded('pcntl')) {
            $this->configurePcntlSignal();
        }
```

The process will run on Windows without that, but might have some implication somewhere else.

**Comment by Ghostscypher** [comment]
Thank you @jeovajr.

**Comment by Ghostscypher** [comment]
@myckhel please advice on the way forward. This issue seems to be related with windows only.

**Comment by vladimirmartsul** [comment]
Hi! I get the same error on the Linux (Gentoo) with PHP-8
```
#php artisan websockets:serve
   Error
  Undefined constant "BeyondCode\LaravelWebSockets\Console\Commands\SIGTERM"
  at vendor/beyondcode/laravel-websockets/src/Console/Commands/StartServer.php:175
    171▕         // signal, it should mark the server as unavailable
    172▕         // to receive new connections, close the current connections,
    173▕         // then stopping the loop.
    174▕
  ➜ 175▕         $this->loop->addSignal(SIGTERM, function () {
    176▕             $this->line('Closing existing connections...');
    177▕
    178▕             $this->triggerSoftShutdown();
    179▕         });
      +14 vendor frames
  15  artisan:37
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
```

**Comment by klilmecha** [comment]
**Problem Solved**
go to this directory => vendor/beyondcode\laravel-websockets\console\commands
open StartServer.php
comment line 82 => //$this->configurePcntlSignal();

**Comment by Ghostscypher** [comment]
Thanks @klilmecha, just wondering, won't this have any repercussions?

**Comment by rennokki** [comment]
In v2, you need `ext-pcntl` and UNIX-based system to run v2.

**Comment by Ghostscypher** [comment]
> In v2, you need `ext-pcntl` and UNIX-based system to run v2.

Unfortunately windows doesn't have ext-pcntl, so my guess for now will be to just run v9 by upgrading composer to require php8. Bit there should be a check for windows and skip that checking part

> **Problem Solved**
> go to this directory => vendor/beyondcode\laravel-websockets\console\commands
> open StartServer.php
> comment line 82 => //$this->configurePcntlSignal();

As shown here

**Comment by rennokki** [comment]
With the attached PR, Windows is now supported, but the soft shutdown does not work with it.

**Comment by Ghostscypher** [comment]
Thank @rennokki, I really appreciate the effort.

**Comment by ProgrammingPlanet** [comment]
i'm getting this thing.....

composer require beyondcode/laravel-websockets
Using version ^2.0@beta for beyondcode/laravel-websockets
./composer.json has been updated
Running composer update beyondcode/laravel-websockets
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

Problem 1
- beyondcode/laravel-websockets[2.0.0-beta.28, ..., 2.0.0-beta.32] require pusher/pusher-php-server ^4.0 -> satisfiable by pusher/pusher-php-server[v4.0.0, ..., v4.1.5].
- beyondcode/laravel-websockets[2.0.0-beta.1, ..., 2.0.0-beta.29] require php ^7.2 -> your php version (8.0.0) does not satisfy that requirement.
- pusher/pusher-php-server[v4.0.0, ..., v4.1.0] require php >=7.1 <7.4 -> your php version (8.0.0) does not satisfy that requirement.
- pusher/pusher-php-server[v4.1.1, ..., v4.1.4] require php ^7.1 -> your php version (8.0.0) does not satisfy that requirement.
- pusher/pusher-php-server v4.1.5 requires ext-curl * -> it is missing from your system. Install or enable PHP's curl extension.
- Root composer.json requires beyondcode/laravel-websockets ^2.0@beta -> satisfiable by beyondcode/laravel-websockets[2.0.0-beta.1, ..., 2.0.0-beta.32].

**Comment by Ghostscypher** [comment]
@ProgrammingPlanet you'll need to first upgrade your pusher server  to a version that supports PHP8 kindly visit their repo to check which version supports this.
Also install ext-curl in your machine, for windows check your PHP.INI file to see if ext-curl has been uncommented, and go Unix check of the extension is enabled by `php -m | curl` if you don't see it then try installing it via apt

**Comment by rennokki** [comment]
@ProgrammingPlanet Latest `beta.32` supports PHP ^8.0 out of the box. You will also need the pusher package set to at least `4.1.5` because it's when they introduced PHP 8.0 support (https://github.com/pusher/pusher-http-php/releases/tag/v4.1.5)

**Comment by amcsi** [comment]
It would be awesome if PHP 8 support would be added to version 1.x. It would make it much easier for our company to transition to PHP 8. Besides, `2.x` has been in beta for a while, and who knows how long it takes until it's stable for use.

**Comment by amcsi** [comment]
I have created a PR to support PHP 8.0. I hope it gets merged: #695
